### PR TITLE
Ahardman portfolio final code review

### PR DESCRIPTION
**Pull request by amandahrdmn**

### CODE REVIEW

File: portfolio_ahardman.html [hunk 1]
```diff
@@ -0,0 +1,123 @@
+<!doctype html>
```
Comment (mporam) [inline portfolio_ahardman.html:1]: file name needs to be index.html please - will explain on friday

File: portfolio_ahardman_stylesheet.css [hunk 2]
```diff
@@ -0,0 +1,339 @@
+/*   default settings   */
+:root{
+	--cyan_light: rgb(0, 204, 191);
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: can you not use CSS variables please. these dont work in IE11. we will show you how to do this properly in a few weeks :)

File: portfolio_ahardman_stylesheet.css [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+	--cyan_dark: rgb(0, 164, 153);
+}
+*{
+	box-sizing: border-box;
+	font-family: 'Segoe UI', Helvetica, Verdana; 
+}
+body{                 	/* removes default padding and margin around the ouside of the entire webpage */
+	font-size: 62.5%;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css:8]: this needs to be on the `<html>` element to work

File: portfolio_ahardman_stylesheet.css [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+	margin: 0;
+	padding: 0;
+	background-image: linear-gradient(to right,rgb(110,110,110),rgb(180,180,180));
+}
+@-ms-viewport{
+	width: device-width;
+}
+a{
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: CSS syntax should be:
```
selector {
```
space before the `{`

Comment (mporam) [reply portfolio_ahardman_stylesheet.css]: also should have a line break between each selector:
```
selector {

}

selector {

}
```

File: portfolio_ahardman_stylesheet.css [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+	text-decoration: none;
+	color:rgb(0,0,0);
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: should always have a space after the `:`

File: portfolio_ahardman_stylesheet.css [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+}
+input:focus{
+	border: none;
+	outline: none;
+}
+svg.logo{
+	z-index: -1;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: negative z-index is generally a bad idea. this actually does nothing in this case

File: portfolio_ahardman_stylesheet.css [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+	fill: none;
+	stroke-linecap: round;
+}
+circle{
+	fill:rgb(0, 164, 153);
+}
+/* ----------------------navbar settings---------------------- */
+.navbar{
+	position:fixed;
+	position: sticky;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: position sticky actually does nothing in this context, just use fixed. will need to use `width: 100%;` to make it work

File: portfolio_ahardman_stylesheet.css [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+	display: flex;
+	justify-content: flex-end;
+	align-items: center;
+	top: 0px;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: dont need `px` on 0 values

File: portfolio_ahardman_stylesheet.css [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+	height: 60px;
+}
+.navbar .link.container{
+	position: relative;
+	left:0;
+	z-index: 99999;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: 99999? 10 would do

File: portfolio_ahardman_stylesheet.css [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+	height: 52px;
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	background-color: rgb(180,180,180);
+	border: 5px solid rgb(0, 204, 191);
+	border-radius: 5px;
+}
+.navbar .link.container a{
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: this selector is a bit too specific, why not just `.navbar a` ?

File: portfolio_ahardman_stylesheet.css [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+	height: 100%;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	margin: 1.8vw;
+	color: rgb(0,0,0);
+	font-size: 1.2rem;
+	transition-duration: 0.8s;
+}
+.navbar .link.container a:visited{
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: same as above

File: portfolio_ahardman_stylesheet.css [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+	height: 100%;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: alot of these properties are already set in the non `:visited` version, you dont have to set them again here, they will inherit

File: portfolio_ahardman_stylesheet.css [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	color: rgb(0,0,0);
+	text-decoration: none;
+}
+.navbar .link.container a:hover{
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: same as above 2 comments

File: portfolio_ahardman_stylesheet.css [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+	color: rgb(255,255,255);
+	transition-duration: 0.3s;
+}
+/* ----------------------main section settings---------------------- */
+main{
+	background-image: linear-gradient(to right,rgb(110,110,110),rgb(180,180,180));
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	height:calc(100vh - 60px);
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: can i use says this doesnt work in IE? sure this works?

File: portfolio_ahardman_stylesheet.css [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+}
+main .container{
+	display: flex;
+	justify-content: center;
+	flex-direction: column;
+	width: 70%;
+	margin:80px 0px 80px 0px;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: this can be shortened to:
`margin: 80px 0;`

File: portfolio_ahardman_stylesheet.css [hunk 16]
```diff
@@ -0,0 +1,339 @@
...
+	padding:30px 0px 30px 0px;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: same as above

File: portfolio_ahardman_stylesheet.css [hunk 17]
```diff
@@ -0,0 +1,339 @@
...
+}
+main h1{
+	font-size: 3.2rem;
+	margin: 0px;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: dont need `px` on 0 values

File: portfolio_ahardman_stylesheet.css [hunk 18]
```diff
@@ -0,0 +1,339 @@
...
+	margin: 2.5vh 0px 0px 0px;
+	padding: 0px;
+	border: 0px;
+}
+/* portfolio section */
+.portfolio{
+	width: 100%;
+	display: flex;
+	justify-content: center;
+	background-color: white;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: can you keep colors consistent, try and use hex whereever possible

File: portfolio_ahardman_stylesheet.css [hunk 19]
```diff
@@ -0,0 +1,339 @@
...
+	margin: 10px;
+	padding-bottom: 10px;
+	display: flex;
+	justify-content: space-around;
+	align-items: flex-end;
+	font-size: 1rem;
+	width: 200px;
+	height: 200px;
+	transition-duration: 0.6s;
+	border: 3px solid rgb(0, 164, 153);
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: hex?

File: portfolio_ahardman_stylesheet.css [hunk 20]
```diff
@@ -0,0 +1,339 @@
...
+.popup{
+	padding: 10px;
+	background-color: rgb(0, 204, 191);
+	border: 1px solid black;
+	border-radius: 5px;
+	font-size: 1rem;
+}
+.build.box1{
+	background-image: url('https://raw.githubusercontent.com/Mayden-Academy/FST-curriculum/master/MAYDEN%20ACADEMY_WEBSAFE_LOGO_06.15.png?token=AORF4IAOEYU56RY3T3FACIK6M6EHG');
+	background-size: 100%;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: background size, repeat and position could just be put on `.build` so you dont have to repeat them

File: portfolio_ahardman_stylesheet.css [hunk 21]
```diff
@@ -0,0 +1,339 @@
...
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	background-color: rgb(0, 204, 191);
+}
+/* ----------------------about me section---------------------- */
+.about_me{
+	padding-top: 150px;
+	padding-bottom: 150px;
+	width: 100%;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: this doesnt do anything

File: portfolio_ahardman_stylesheet.css [hunk 22]
```diff
@@ -0,0 +1,339 @@
...
+	align-items: center;
+	margin-left: 2vw;
+}
+form{
+	display: flex;
+	align-items: center;
+	width: 100%;
+}
+.email_input{
+	margin-right: 1vw;
```
Comment (mporam) [inline portfolio_ahardman_stylesheet.css]: generally using vw for margins and padding is a bad idea. suggest changing to px

**Review by mporam — CHANGES_REQUESTED** [review]

File: portfolio_ahardman_stylesheet.css (hunk 14 above)
Comment (amandahrdmn) [reply portfolio_ahardman_stylesheet.css]: Expanding to fill the viewport doesn't occur in IE11. It's still readable - should I remove anyway?

**Review by mporam — APPROVED** [review]

**Pull request merged**